packages/dds/sequence/src/sequence.ts

```diff
@@ -621,18 +621,24 @@ export abstract class SharedSegmentSequence<T extends ISegment>
 							m.minimumSequenceNumber < collabWindow.minSeq ||
 							m.referenceSequenceNumber < collabWindow.minSeq ||
 							m.sequenceNumber <= collabWindow.minSeq ||
-							m.sequenceNumber <= collabWindow.currentSeq
+							m.sequenceNumber < collabWindow.currentSeq ||
```

Comment:
if clientSeq is non-optional, i think the condition can be simplified:
```suggestion
							(m.sequenceNumber <= collabWindow.currentSeq &&
									m.clientSequenceNumber <= collabWindow.currentClientSeq)
```

Reply:
actually, i think i just found a problem. this code needs to handle both local and remote ops, you should test both

Reply:
all that other checks were invariant for all ops. the best check just might be `m.sequenceNumber <= collabWindow.currentSeq` and not touch clientSequenceNumber

Reply:
i guess its not local vs remote, but you can have ops from two different clients, which can have different clients seqs. at least for old client seqs, i don't really understand how they have changed

Reply:
I don't think it needs to explicitly handle both local and remote ops. The check only happens if the sequence number matches, and that can only happen with ops from the same client.
That's why I don't think we can combine checking `<` and `===` into one. We have to do them seperately.

The simplest this check could get with `currentClientSeq` required is:
```
m.sequenceNumber < collabWindow.currentSeq ||
(m.sequenceNumber === collabWindow.currentSeq
	&& m.clientSequenceNumber <= collabWindow.currentClientSeq)
```